### DISCUSSION TITLE
XRender 2.0

### DISCUSSION THREAD

**Discussion by siyi98** [alibaba/x-render#770] [Ideas]
### What's new in XRender2.0 ? 👀
1. 支持跨平台渲染（PC、Mobile、小程序）
2. 提高组件渲染性能、易用性、稳定性
3. ...

**Comment by Jarod-Xie** [comment]
table-render 支持完整的schema配置

**Comment by Jarod-Xie** [comment]
表单设计器现在只能用来生成表单，能不能扩展一下，弄成页面设计器，可以拖拽表单、表格、图表，布局容器，生成页面

**Comment by MoYuM** [comment]
1. 支持增量更新。
2. 使用原生 <form> 标签构建表单（要不然没有 auto complete）。
3. 尽量保证每个字段的功能是单一的，减少功能上的重叠。比如现在 `type`，`format`，`widget` 三个字段都会影响最终使用的是哪个组件，建议只用 `widget` 这一个字段。然后 `type` 除了影响组件，还会影响校验。建议分开处理。如果是语法糖的话，一个字段也只能是一个功能的语法糖。
4. 更好的 schema 校验，可以考虑引入新的库。
5. 增加表单模式，比如 Modal 模式，Drawer 模式，表单筛选模式，SOP模式等。
6. 错误样式跟 antd 对齐。目前只有一个简单的错误信息的提示，但其实不少 antd 组件本身是有错误状态的，应该加上。
7.  这一点有待商榷，是否要继续坚持 JSON Schema 的格式？我觉得除非有特别的要求，否则没有必要再坚持。因为：
     - 我们需要在 schema 中引入函数来增强功能，比如校验。
     - schema 能更简洁，比如 `schema.type`，比如 `list.items.type` 这些必填项，始终都是那些固定值，完全可以去掉。
     - 语义更加准确，为什么用 properties（属性） 代表表单项？。
     - json 格式可能更适合服务端下发的场景，但我们完全可以想办法兼容 json 格式的同时，让 FR 支持更多内容。
8. 支持栅格布局，span、offset 之类的。
9. 更多的测试用例，保证每次发布时的质量。

**Reply by siyi98** [reply]
增量更新是什么意思？

**Reply by MoYuM** [reply]
哦，就是输入数据的时候只更新必要的组件，比如我正在一个 Input 中输入点什么，那正常情况下应该只有这个 Input 会重新渲染。像 antd 
form 那样

**Reply by siyi98** [reply]
明白了

**Comment by lyhper** [comment]
表单设计器做深：联动、分步表单、弹窗表单
横向场景拓展：表格、图表、详情

**Reply by siyi98** [reply]
@lyhper 图表这块的需求，可以详细说说吗

**Comment by vigle** [comment]
XRender2.0  大概什么时候推出

**Comment by yuezeliu** [comment]
TableRender 结合 FormRender做行内编辑。

**Comment by siyi98** [comment]
- #851 
- #854 
- #820

**Comment by xingdongyu1994** [comment]
1. 校验可以（如antd）支持动态的校验
2. 布局上有栅格布局是最好的（可以提高设计的还原度）
3. 设计器上可以支持灵活的定制， 感觉定制能力贼差
     可以对组件样式的设置
     组件列表区面板样式
     属性配置区面板样式
     对联动的配置（这个感觉比较重要）
     。。。。
     这些都搞不定

**Comment by dengyao** [comment]
请问下移动端支持什么时候能够使用

**Comment by wenjianzhang** [comment]
组件中 宽度能否增加 md xs 等antd pro 中定义支持。
